spec/models/email_address_spec.rb

```diff
@@ -89,25 +89,80 @@
     end
   end
 
-  describe '#gov_or_mil?' do
-    subject(:result) { email_address.gov_or_mil? }
+  describe '#fed_or_mil_email?' do
+    subject(:result) { email_address.fed_or_mil_email? }
 
-    context 'with an email domain ending in anything other than .gov or .mil' do
-      let(:email) { 'example@example.com' }
+    context 'with an email domain that is a fed email' do
+      before do
+        allow(IdentityConfig.store).to receive(:use_fed_domain_class).and_return(false)
+      end
+      let(:email) { 'example@example.gov' }
+
+      it { expect(result).to eq(true) }
+    end
+
+    context 'with an email that is a mil email' do
+      let(:email) { 'example@example.mil' }
+
+      it { expect(result).to be_truthy }
+    end
+
+    context 'with an email that is not a mil or fed email' do
+      before do
+        allow(IdentityConfig.store).to receive(:use_fed_domain_class).and_return(true)
+      end
+
+      let(:email) { 'example@bad.gov' }
 
-      it { expect(result).to eq(false) }
+      it { expect(result).to be_falsey }
```

Comment:
Per previous discussion, we can set all `be_falsey` to `eq(false)` and `be_truthy` to `eq(true)` ?

```suggestion
      it { expect(result).to eq(true) }
```